tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/AnnotationRuntimeTest.java

```diff
@@ -47,153 +46,145 @@
-//        Assert.assertEquals(TypeChecker.getType(annotValue).getTag(), TypeTags.MAP_TAG);
-//
-//        mapValue = (MapValueImpl<BString, Object>) annotValue;
-//        Assert.assertEquals(mapValue.size(), 2);
-//        Assert.assertEquals(mapValue.get(StringUtils.fromString("q")).toString(), "hello");
-//        Assert.assertEquals(mapValue.get(StringUtils.fromString("r")).toString(), "world");
-//    }
-//
-//    public void testRecordTypeAnnotationReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testRecordTypeAnnotationReadonlyValueEdit");
-//    }
-//
-//    public void testAnnotationOnObjectTypeReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testAnnotationOnObjectTypeReadonlyValueEdit");
-//    }
-//
-//    public void testAnnotationOnFunctionTypeReadonlyValueEdit() {
-//        BRunUtil.invoke(resultAccessNegative, "testAnnotationOnFunctionTypeReadonlyValueEdit");
-//    }
-//
-//    public void testReadonlyTypeAnnotationAttachment() {
-//        BRunUtil.invoke(readOnlyValues, "testReadonlyTypeAnnotationAttachment");
-//    }
+    private CompileResult resultOne;
+    private CompileResult resultTwo;
+    private CompileResult resultThree;
+    private CompileResult resultFour;
+    private CompileResult resultAccessNegative;
+    private CompileResult readOnlyValues;
+
+    @BeforeClass
+    public void setup() {
+        resultOne = BCompileUtil.compile("test-src/annotations/annot_access.bal");
+        Assert.assertEquals(resultOne.getErrorCount(), 0);
+
+        resultTwo = BCompileUtil.compile("test-src/annotations/annot_access_with_source_only_points.bal");
+        Assert.assertEquals(resultTwo.getErrorCount(), 0);
+
+        resultThree = BCompileUtil.compile("test-src/annotations/annotations_constant_propagation.bal");
+        Assert.assertEquals(resultThree.getErrorCount(), 0);
+
+        resultFour = BCompileUtil.compile("test-src/annotations/annot_availability.bal");
+        Assert.assertEquals(resultFour.getErrorCount(), 0);
+
+        resultAccessNegative = BCompileUtil.compile("test-src/annotations/annotation_access_negative.bal");
+
+        readOnlyValues = BCompileUtil.compile("test-src/annotations/annotation_readonly_types.bal");
+    }
+
+    @Test(description = "test accessing source only annots at runtime, the annots should not be available",
+            dataProvider = "annotAccessWithSourceOnlyPointsTests")
+    public void testSourceOnlyAnnotAccess(String testFunction) {
+        BValue[] returns = BRunUtil.invoke(resultTwo, testFunction);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertSame(returns[0].getClass(), BBoolean.class);
+        Assert.assertTrue(((BBoolean) returns[0]).booleanValue());
+    }
+
+    @Test(description = "Test if constants used in annotation are replaced on constant propagation phase",
+          enabled = false)
+    public void testConstantPropagationOnAnnotation() {
```

Comment:
Any update on if we can do this now?

Reply:
Still, I didn't find a way to enable this test. Shall I enable this with https://github.com/ballerina-platform/ballerina-lang/issues/28699